ui/src/app/workflows/components/workflows-row/workflows-row.tsx

```diff
@@ -87,7 +87,7 @@ export function WorkflowsRow(props: WorkflowsRowProps) {
                     <div className='columns small-1'>{isArchivedWorkflow(wf) ? 'true' : 'false'}</div>
                     {(props.columns || []).map(column => {
                         // best not to make any assumptions and wait until this data is filled
-                        const value = wf?.metadata?.labels?.[column.key] ?? 'unknown';
+                        const value = wf?.metadata?.annotations?.[column.key] ?? wf?.metadata?.labels?.[column.key] ?? 'unknown';
```

Comment:
we should use `column.type` directly here
```suggestion
                        const value = (column.type === "label" ? wf?.metadata?.labels?.[column.key] : wf?.metadata?.annotations?.[column.key]) ?? 'unknown';
```

I'm not the biggest fan of large one-liners though

Reply:
I'm not a fan of this. You feel strongly about this? Am I missing a possible issue without the type check explicitly?

Reply:
> Am I missing a possible issue without the type check explicitly?

Yes, for example, you can capture a label even when the user/operator specified the `annotation` type

Reply:
```suggestion
                        const keys = column.type === "label" ? wf?.metadata?.labels : wf?.metadata?.annotations;
                        const value = keys?.[column.key] ?? 'unknown';
```

although I'd prefer we specify `label` and `annotation` explicitly too.... hmmm...

```suggestion
                        const keys =  wf?.metadata?.[`${column.type}s`];
                        const value = types?.[column.key] ?? 'unknown';
```

well that could be one-lined as dynamic af:

```suggestion
                        const value = wf?.metadata?.[`${column.type}s`]?.[column.key] ?? 'unknown';
```

Reply:
Ah, nice catch.  Example for other readers.  We have the following in the workflow:

```yaml
labels:
  workflows.argoproj.io/custom-column: test-label
```

and our configmap is only looking for annotations.

Gonna look at

Reply:
did you not want to use the dynamic version in my last suggestion [above](https://github.com/argoproj/argo-workflows/pull/12421#discussion_r1437976854)?

Reply:
I'd do one of the following for clean up

(yours but rename `keys` to `metadata`)

```javascript
const metadata = column.type === "label" ? wf?.metadata?.labels : wf?.metadata?.annotations;
const value = metadata?.[column.key] ?? 'unknown';
```

or VERY EXPLICIT

```javascript
let value = 'unknown';
if (column.key === 'labels') {
    value = wf?.metadata?.labels?.[column.key] ?? 'unknown';
}
if (column.key === 'annotations') {
    value = wf?.metadata?.annotations?.[column.key] ?? 'unknown';
}
```

Not a fan of of the `${column.type}s` stuff. 😢